bin/growpart

```diff
@@ -873,6 +891,19 @@ while [ $# -ne 0 ]; do
 			Usage
 			exit 0
 			;;
+		--free-percent|--free-percent=*)
+			if [ "${cur#--free-percent=}" != "$cur" ]; then
+				next="${cur#--free-percent=}"
+			else
+				shift
+			fi
+			if [ "$next" -gt 0 ] 2>/dev/null &&
+				[ "$next" -lt 100 ] 2>/dev/null; then
```

Comment:
In practice I don't think we can really accept 100 percent here. It's probably overkill, but I'm wondering if we should do a calculation of max_percent calculation based on the percentage represented by pt_end? We could alternately limit it to a hard-coded 90 percent or something? and document that acceptable range in the usage output.

Reply:
> In practice I don't think we can really accept 100 percent here

It doesn't, the logic is ">0 AND <100" so 1-99%.

I did think about having a smaller arbitary range, e.g. 5-10% or 5-15%, instead.

Reply:
I think this is good as it is, checking that `0 < x < 100`. It's the most basic sanity check, which I think it's mostly useful to catch malformed percentages (e.g. with decimals).

Reply:
Works for me thanks @paride. I think we we create an arbitrarly low MAX there'll be someone who is trying to test some funky config that we might want to allow at some point. All good as is, just wanted a little discussion on this. Thanks for the context @dermotbradley

Reply:
@blackboxsw A realistic scenario when a large percentage figure might be used that I'm familiar with: using cloud-init on a Raspberry Pi which uses a SDcard for the OS. It is hard to find SDcards smaller than 4GB in size these days but the Alpine base disk image I build for RPIs is only 200-300MB or so in size so, when installed on a 4GB (or larger) SDcard even with data/logfile use I might wish to never partition more than 50% of the SDcard to (hopefully) extend its lifetime.